torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -328,5 +350,35 @@ c10::intrusive_ptr<ProcessGroup::Work> barrier(
   return op.call(process_group, opts.device_ids, opts.timeout.count());
 }
 
+c10::intrusive_ptr<ProcessGroup::Work> send(
+    const c10::intrusive_ptr<ProcessGroup>& process_group,
+    at::TensorList tensors,
+    int64_t dstRank,
+    int64_t tag) {
+  static auto op = c10::Dispatcher::singleton()
+                       .findSchemaOrThrow("c10d::send", "")
+                       .typed<c10::intrusive_ptr<::c10d::ProcessGroup::Work>(
+                           at::TensorList,
+                           const c10::intrusive_ptr<::c10d::ProcessGroup>&,
+                           int64_t,
+                           int64_t)>();
+  return op.call(tensors, process_group, dstRank, tag);
+}
+
+c10::intrusive_ptr<ProcessGroup::Work> recv(
```

Comment:
just noticed we have this `recv` and `recv_` two separate APIs (above), what's the difference between these two and should we merge?

Reply:
The above recv_ is the implementation of the function schema. This one is the C++ API for users to call the op. recv_ follows aten ops's conventions while recv follows c10d's conventions.